pkg/kubelet/metrics/collectors/resource_metrics.go

```diff
@@ -54,7 +54,21 @@ var (
 		metrics.ALPHA,
 		"")
 
-	resouceScrapeResultDesc = metrics.NewDesc("scrape_error",
+	podCPUUsageDesc = metrics.NewDesc("pod_cpu_usage_seconds_total",
+		"Cumulative cpu time consumed by the pod in core-seconds",
+		[]string{"pod", "namespace"},
+		nil,
+		metrics.ALPHA,
+		"")
+
+	podMemoryUsageDesc = metrics.NewDesc("pod_memory_working_set_bytes",
+		"Current working set of the pod in bytes",
+		[]string{"pod", "namespace"},
```

Comment:
How is this different from `container_memory_working_set_bytes{pod="X",namespace="Y",container=""}` (which should be working set of the pod)?

Reply:
Not all processes associated with running a pod are within the container cgroups themselves, but instead within the pod cgroup created by kubelet. For example, conmon in the standard/crio case.

When looking at sandboxed runtimes, this is even more significant, as the pod cgroup will include processes like the VMM, vCPU threads, IO threads, etc.

Reply:
`container_memory_working_set_bytes{pod="X",namespace="Y",container=""}` is the pod cgroup measurement today of memory.  This should not be sum of container usage, but is the direct cgroup measured value (and has been for quite a few releases).  Crio should account for conmon in the pod cgroup (although I know recently we changed some stuff around - @mrunalp what was the outcome here?)

Here's a 1.19 cluster with crio from systemd-cgls:

```
  │ ├─kubepods-burstable-podb7e40d48_a9da_4b45_bc36_d5d1a6cc2fe2.slice
  │ │ ├─crio-44974e038bd96557f71f44779b0a1603bb02b42494ee8ffce3aae974fc40264f.scope
  │ │ │ ├─4076273 /bin/sh /var/lib/tuned/bin/run start
  │ │ │ ├─4076299 openshift-tuned -v=0
  │ │ │ └─4076484 /usr/libexec/platform-python -Es /usr/sbin/tuned --no-dbus
  │ │ ├─crio-conmon-d3ed68c63a2c3dfadc1082574d4dae59946120a7c49056d18f970b75ca520e43.scope
  │ │ │ └─4076177 /usr/libexec/crio/conmon -b /var/run/containers/storage/overlay-containers/d3ed68c63a2c3dfadc1082574d4dae59946120a7c49056d18f970b75ca520e43/userdata ->
```

```
$ cat /sys/fs/cgroup/memory/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-podb7e40d48_a9da_4b45_bc36_d5d1a6cc2fe2.slice/memory.memsw.usage_in_bytes
65708032
```

From prometheus 

```

  | 58855424
-- | --
container_memory_usage_bytes{endpoint="https-metrics",id="/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-podb7e40d48_a9da_4b45_bc36_d5d1a6cc2fe2.slice",instance="10.0.153.142:10250",job="kubelet",metrics_path="/metrics/cadvisor",namespace="openshift-cluster-node-tuning-operator",node="ip-10-0-153-142.ec2.internal",pod="tuned-cg8rv",service="kubelet"}
```

`container=""` shows the pod cgroup usage correctly on 1.19 at least.

Reply:
@smarterclayton We ship with the conmon scope nested under the pod cgroup slice. We have an option to move it to a dedicated cgroup slice outside the pod, but don't use it in OpenShift.